site/content/en/docs/Advanced/high-availability-agones.md

```diff
@@ -0,0 +1,71 @@
+weight: 20
+description: >
+  Allows controllers to be strictly for control loops; feature separates extensions and webhooks out of your controllers
+---
+
+## High Availability for Agones Controller
+
+The `agones-controller` pod is currently not only used for control loops of game server related resources, but also include other services such as webhooks and allocation extension srvices. High availability agones mean as opposed to having controller host everything mentioned, we will add additional `extensions` pods that exposes the webhook and allocations service. Thus, making `agones-controller` pod strictly for control loops. Read more about pod disruption [here](https://agones.dev/site/docs/advanced/controlling-disruption/)
+
+{{< alpha title="`split controller and extensions` API" gate="SplitControllerAndExtensions" >}}
+
+{{< alert title="GKE Autopilot Clusters" color="warning" >}}
+In order to use GKE Autopilot clusters with Agones, this feature flag `SplitControllerAndExtensions` must be enabled along with `SafeToEvict` (should already be enabled). 
+
+More info on `SafeToEvict` [here](https://agones.dev/site/docs/advanced/controlling-disruption/#disruption-in-kubernetes)
+{{< /alert >}}
+
+## Extension Pod Configrations 
+
+Current setup has `extensions` pods inherit values through the `controller` values within the `values.yaml` through **yaml inheritance**. This means that changes made to the `controller` configurations within the file will have a direct effect on the `extensions` values. Essentially acts as a copy & paste. To override the configurations within `values.yaml`, declare and initialize the desired key and value within the `extensions` section within `values.yaml`. For example, lets change the `apiServerQPS` field value inhertied for `extensions` to 600 instead of the original 400:  
+
+```
+  controller: &example
+    numWorkers: 100
+    apiServerQPS: 400
+    apiServerQPSBurst: 500
+    http:
+      port: 8080
+...
+  extensions:
+    <<: *example
+    apiServerQPS: 600
+```
+
+Another important note is that `helm --set` sets the variable after the yaml inheritance, helm resolves the chart for values and apply the changes from `--set`. Changing `controller` value through this method will not change the values that `extensions` inherited. The same applies for when `--values` or `-f` is used. 
+
+To change `controller.numWorkers` to 200 from 100 values and through the use of `helm --set`, add the follow to the `helm` command:
+
+{{< alert color="warning" >}} Important: This will not have any effect on any `extensions` values! {{< /alert >}}
+```
+ ...
+ --set agones.controller.numWorkers=200
+ ...
+```
+
+## Feature Enabled 
+
+The `SplitControllerAndExtensions` feature creates sperate pods for extensions, currently the replication is set to `2` but it can definitely be changed through `values.yaml` or `helm --set`. Extensions pods do have PDB (PodDisruptionBudget) and has the `minAvailable` set to `1`.
```

Comment:
Reword this.. maybe:

---
When `SplitControllerAndExtensions` is enabled, what was previously a single `agones-controller` pod is deployed as `agones-controller` and 2 `agones-extensions` pods, like:

```
NAME                                 READY   STATUS    RESTARTS   AGE
agones-controller-fbf944f4-vs9xx     1/1     Running   0          23h
agones-extensions-5648fc7dcf-hm6lk   1/1     Running   0          23h
agones-extensions-5648fc7dcf-qbc6h   1/1     Running   0          23h
[...]
```

The number of `agones-extensions` replicas can be set using helm variable _blah_(link to helm variable reference).

We expect the aggregate memory consumption of the pods will be slightly higher than the previous singleton pod, but as the responsibilities are now split across the pods, the aggregate CPU consumption should be similar.
---

(I would drop the mention of the PDB - we don't doc every helm variable in the guides. Do link the helm reference, though.)